File: dart/lib/bebop_handlers/ghost_move.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class GhostMoveMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    var decodedMessage = GhostMoveMsg.decode(message);
    for (var pos in decodedMessage.ghostPositions.entries) {
      GameState.instance.ghostPositions
          .update(pos.key, (value) => pos.value, ifAbsent: () => pos.value);
    }
  }
}

File: typescript/src/bebop_handlers/ghostMove.ts
import GameState from '../classes/gameState';
import { GhostMoveMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class GhostMoveMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = GhostMoveMsg.decode(message);
        for (var pos of message.ghostPositions.entries()) {
            GameState.Instance.ghostPositions.set(pos[0], pos[1]);
        }
    }
}